### PR TITLE
Ebook Online reading

### PR DESCRIPTION
**Pull request by ragusa87**
### Allows you to read ebook online (not only cbr/cbz)

* Support for reading epub online (not tested with mobi)
* Show Book UUID in Ux
* Use Vue.js and <https://github.com/jinhuan138/vue-book-reader/> (based on foliate-js)

### CODE REVIEW

File: templates/book/index.html.twig [hunk 1]
```diff
@@ -128,11 +128,28 @@
...
+                <tr>
+                    <th>{{ 'Number of pages'|trans }}</th>
+                    <td>
+                        {{ component('InlineEditBook', {'book':book, 'field':'pageNumber'}) }}
+                    </td>
+                </tr>
+                <tr>
+                    <th>{{ 'UUID'|trans }}</th>
+                    <td>{{ book.uuid }}</td>
+                </tr>
```
Comment (SergioMendolia) [inline templates/book/index.html.twig:143]: is it really necessary to display it everywhere? 99% of users won't care

File: templates/book/reader-files-epub.html.twig [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+{% endblock %}
+
+{% block stylesheets %}
+    {{ encore_entry_link_tags('read-ebook') }}
+    <style>
+        body
+        {
+            background-color: #000;
+            color: #fff;
+        }
```
Comment (SergioMendolia) [inline templates/book/reader-files-epub.html.twig:16]: https://media3.giphy.com/media/7SF5scGB2AFrgsXP63/giphy.gif

File: package.json [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
     "bootstrap": "^5.3.1",
     "core-js": "^3.23.0",
+    "eslint": "^8.57.0",
+    "eslint-plugin-vue": "^9.23.0",
+    "prettier": "^3.2.5",
     "regenerator-runtime": "^0.13.9",
     "sass": "^1.62.0",
     "sass-loader": "^13.2.2",
     "tom-select": "^2.2.2",
+    "vite": "^5.3.1",
```
Comment (SergioMendolia) [inline package.json:27]: why?

Comment (ragusa87) [reply package.json:27]: To be frank I don't know, I followed a "getting started with Vue tutorial" and copy-pasted the dependencies. I will try to get rid of it

Comment (ragusa87) [reply package.json]: I removed it without issue

File: templates/book/index.html.twig (hunk 1 above)
Comment (ragusa87) [reply templates/book/index.html.twig:142]: I added `{% if app.debug }` around it.

File: templates/book/reader-files-epub.html.twig (hunk 2 above)
Comment (ragusa87) [reply templates/book/reader-files-epub.html.twig]: I moved that to scss, and injected a class instead. 

I also took care of changing the vue component to better support theming.

**Comment by SergioMendolia** [comment]
Vite is a modern alternative to web pack so I'm pretty sure it’s not used   
Le mar. 6 août 2024 à 21:01, Laurent Constantin ***@***.***&gt; a écrit :  @ragusa87 commented on this pull request.



In package.json:
&gt;      "regenerator-runtime": "^0.13.9",
     "sass": "^1.62.0",
     "sass-loader": "^13.2.2",
     "tom-select": "^2.2.2",
+    "vite": "^5.3.1",

To be frank I don't know, I followed a "getting started with Vue tutorial" and copy-pasted the dependencies. I will try to get rid of it

**Review by SergioMendolia — APPROVED** [review]

**Pull request merged**